newIDE/app/src/Leaderboard/useLeaderboardReplacer.js

```diff
@@ -0,0 +1,415 @@
+                primary
+                key="register-game-later"
+                label={<Trans>Register later</Trans>}
+                onClick={onClose}
+              />,
+              <DialogPrimaryButton
+                primary
+                key="register-game-now"
+                label={<Trans>Register now</Trans>}
+                onClick={onTriggerReplace}
+              />,
+            ]
+          : [
+              <FlatButton
+                primary
+                key="login-later"
+                label={<Trans>Login later</Trans>}
+                onClick={onClose}
+              />,
+              <DialogPrimaryButton
+                label={<Trans>Login now</Trans>}
+                primary
+                onClick={() => onLogin()}
+                key="login-now"
+              />,
+            ]
+      }
+      title={<Trans>Leaderboards detected!</Trans>}
+      noMargin
+      open
+      maxWidth="sm"
+      onRequestClose={onClose}
+    >
+      <Line>
+        <ColumnStackLayout>
+          <Text>
+            <Trans>
+              We detected leaderboards that are not yet valid. Registering this
+              project with your GDevelop account would allow us to create them
+              in your stead so that your project is ready to use and publish
+              with minimal efforts.
+            </Trans>
+          </Text>
+          <Text>
+            <Trans>
+              You can still do this operation manually later from the
+              leaderboard configuration panel.
+            </Trans>
+          </Text>
+        </ColumnStackLayout>
+      </Line>
+    </Dialog>
+  );
+};
+
+type LeaderboardReplacerProgressDialogProps = {|
+  erroredLeaderboards: Array<{ leaderboardId: string, error: Error }>,
+  onRetry: ?() => void,
+  onAbandon: ?() => void,
+  progress: number,
+|};
+
+export const LeaderboardReplacerProgressDialog = ({
+  erroredLeaderboards,
+  onRetry,
+  onAbandon,
+  progress,
+}: LeaderboardReplacerProgressDialogProps) => {
+  const hasErrors = erroredLeaderboards.length > 0;
+
+  return (
+    <Dialog
+      actions={[
+        onRetry ? (
+          <DialogPrimaryButton
+            label={<Trans>Retry</Trans>}
+            primary
+            onClick={onRetry}
+            key="retry"
+          />
+        ) : null,
+        <FlatButton
+          label={
+            onAbandon ? <Trans>Abandon</Trans> : <Trans>Please wait...</Trans>
+          }
+          disabled={!onAbandon}
+          onClick={onAbandon}
+          key="close"
+        />,
+      ]}
+      cannotBeDismissed={!hasErrors}
+      noMargin
+      open
+      maxWidth="sm"
+    >
+      <Line>
+        <ColumnStackLayout expand>
+          <Text>
+            {hasErrors && progress === 100 ? (
+              <Trans>
+                There were errors when duplicating leaderboards for the project.
+              </Trans>
+            ) : (
+              <Trans>Leaderboards are duplicated...</Trans>
```

Comment:
duplicating?
Or `Please wait while we are creating leaderboards for your game`

Reply:
I don't think we should tell the user the details of the leaderboard implementation. For them, we are just creating leaderboard, mentioning a duplication could confuse them.

Reply:
Correct, we can just say "Preparing the leaderboard for your game"/"Please wait while we create leaderboards for your game..."